src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/RecallDocumentService.kt

```diff
@@ -0,0 +1,36 @@
+package uk.gov.justice.digital.hmpps.managerecallsapi.service
+
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.managerecallsapi.db.RecallDocument
+import uk.gov.justice.digital.hmpps.managerecallsapi.db.RecallDocumentCategory
+import uk.gov.justice.digital.hmpps.managerecallsapi.db.RecallRepository
+import uk.gov.justice.digital.hmpps.managerecallsapi.storage.S3Service
+import java.util.UUID
+import javax.persistence.EntityNotFoundException
+
+@Service
+class RecallDocumentService(
+  @Autowired private val s3Service: S3Service,
+  @Autowired private val recallRepository: RecallRepository
+) {
+
+  private val log = LoggerFactory.getLogger(this::class.java)
+
+  fun addDocumentToRecall(recallId: UUID, documentBytes: ByteArray, documentCategory: RecallDocumentCategory) {
+    val recall = try {
+      recallRepository.getById(recallId)
+    } catch (e: EntityNotFoundException) {
+      throw RecallNotFoundError("No recall found with ID '$recallId'", e)
+    }
+
+    s3Service.uploadFile(documentBytes)
```

Comment:
Need to use the UUID returned in the S3BulkResponseEntity when saving the document don't we?

Reply:
yes - good catch - thanks